From 37713b42a501892469234b90454731d8d8b7a3e6 Mon Sep 17 00:00:00 2001
From: Sergey Poznyakoff <gray@gnu.org>
Date: Mon, 17 Aug 2020 20:34:21 +0300
Subject: Bugfix (complements fd9a86d37b)

* mda/lib/mailquota.c (sql_retrieve_quota): Fix a leftover use of
get_size.
---
 mda/lib/mailquota.c | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/mda/lib/mailquota.c b/mda/lib/mailquota.c
index 7b7dd52d9..af719a591 100644
--- a/mda/lib/mailquota.c
+++ b/mda/lib/mailquota.c
@@ -270,16 +270,10 @@ sql_retrieve_quota (char *name, mu_off_t *quota)
 	}
       else if (tmp == NULL || tmp[0] == 0 || mu_c_strcasecmp (tmp, "none") == 0)
 	rc = RETR_UNLIMITED;
-      else
+      else if (get_quota (quota, tmp)) 
 	{
-	  char *p;
-	  
-	  if (get_size (tmp, quota, &p))
-	    {
-	      mu_error (_("bogus mailbox quota for `%s' (near `%s')"),
-			name, p);
-	      *quota = groupquota;
-	    }
+	  *quota = groupquota;
+	  rc = RETR_OK;
 	}
     }
   
-- 
cgit v1.2.1

